cpp/src/arrow/adapters/orc/adapter.h

```diff
@@ -56,55 +66,114 @@ class ARROW_EXPORT ORCFileReader {
   ///
   /// \param[out] out the returned Table
+  ARROW_DEPRECATED("Deprecated in 6.0.0.")
   Status Read(std::shared_ptr<Table>* out);
 
+  /// \brief Read the file as a Table
+  ///
+  /// The table will be composed of one record batch per stripe.
+  ///
+  /// \return the returned Table
+  Result<std::shared_ptr<Table>> Read();
+
   /// \brief Read the file as a Table
   ///
   /// The table will be composed of one record batch per stripe.
   ///
   /// \param[in] schema the Table schema
   /// \param[out] out the returned Table
+  ARROW_DEPRECATED("Deprecated in 6.0.0.")
   Status Read(const std::shared_ptr<Schema>& schema, std::shared_ptr<Table>* out);
 
+  /// \brief Read the file as a Table
+  ///
+  /// The table will be composed of one record batch per stripe.
+  ///
+  /// \param[in] schema the Table schema
+  /// \return the returned Table
+  Result<std::shared_ptr<Table>> Read(const std::shared_ptr<Schema>& schema);
+
   /// \brief Read the file as a Table
   ///
   /// The table will be composed of one record batch per stripe.
   ///
   /// \param[in] include_indices the selected field indices to read
   /// \param[out] out the returned Table
+  ARROW_DEPRECATED("Deprecated in 6.0.0.")
   Status Read(const std::vector<int>& include_indices, std::shared_ptr<Table>* out);
 
+  /// \brief Read the file as a Table
+  ///
+  /// The table will be composed of one record batch per stripe.
+  ///
+  /// \param[in] include_indices the selected field indices to read
+  /// \return the returned Table
+  Result<std::shared_ptr<Table>> Read(std::vector<int>& include_indices);
+
   /// \brief Read the file as a Table
   ///
   /// The table will be composed of one record batch per stripe.
   ///
   /// \param[in] schema the Table schema
   /// \param[in] include_indices the selected field indices to read
   /// \param[out] out the returned Table
+  ARROW_DEPRECATED("Deprecated in 6.0.0.")
   Status Read(const std::shared_ptr<Schema>& schema,
               const std::vector<int>& include_indices, std::shared_ptr<Table>* out);
 
+  /// \brief Read the file as a Table
+  ///
+  /// The table will be composed of one record batch per stripe.
+  ///
+  /// \param[in] schema the Table schema
+  /// \param[in] include_indices the selected field indices to read
+  /// \return the returned Table
+  Result<std::shared_ptr<Table>> Read(const std::shared_ptr<Schema>& schema,
+                                      const std::vector<int>& include_indices);
+
   /// \brief Read a single stripe as a RecordBatch
   ///
   /// \param[in] stripe the stripe index
   /// \param[out] out the returned RecordBatch
+  ARROW_DEPRECATED("Deprecated in 6.0.0.")
   Status ReadStripe(int64_t stripe, std::shared_ptr<RecordBatch>* out);
 
+  /// \brief Read a single stripe as a RecordBatch
+  ///
+  /// \param[in] stripe the stripe index
+  /// \return the returned RecordBatch
+  Result<std::shared_ptr<RecordBatch>> ReadStripe(int stripe);
```

Comment:
Is there a reason to pass the stripe index as `int` instead of `int64_t`? If not, le's remain consistent.